test/instances/ezETH/EzETHHyperdrive.t.sol

```diff
@@ -0,0 +1,1110 @@
+        // TVL to get interest:
+        //
+        //  sharePriceBefore * adjustedVariableRate = sharePriceAfter
+        //
+        //  becomes:
+        //
+        //  (tvlBefore / ezETHSupply) * adjustedVariableRate = tvlAfter / ezETHSuuply
+        //
+        //  tvlBefore * adjustedVariableRate = tvlAfter
+        //
+        //  Since the ezETHSupply is held constant when we advanceTime.
+
+        (, uint256 totalTVLBefore, ) = getSharePrice();
+        vm.warp(block.timestamp + timeDelta);
+
+        // Accrue interest in Renzo. Since the share price is given by
+        // `RESTAKE_MANAGER.calculateTVLs() / EZETH.totalSupply()`, we can simulate the
+        // accrual of interest by adding to the balance of the DepositQueue contract.
+        // RestakeManager adds the balance of the DepositQueue to totalTVL in calculateTVLs()
+        uint256 adjustedVariableRate = uint256(variableRate).mulDivDown(
+            POSITION_DURATION_2_WEEKS,
+            365 days
+        );
+        uint256 ethToAdd = totalTVLBefore.mulDown(adjustedVariableRate);
+        if (variableRate >= 0) {
+            vm.startPrank(address(RESTAKE_MANAGER));
+            vm.deal(address(RESTAKE_MANAGER), ethToAdd);
+            // use this method because no fees are taken
+            DEPOSIT_QUEUE.depositETHFromProtocol{ value: ethToAdd }();
+        } else {
+            // NOTE: can't support subtracting eth when depositQueue has a zero balance.
+            vm.deal(
+                address(DEPOSIT_QUEUE),
+                address(DEPOSIT_QUEUE).balance - ethToAdd
+            );
+        }
+    }
+
+    struct AccountBalances {
+        uint256 ezethBalance;
+        uint256 ETHBalance;
+    }
+
+    function getAccountBalances(
+        address account
+    ) internal view returns (AccountBalances memory) {
+        return
+            AccountBalances({
+                ezethBalance: EZETH.balanceOf(account),
+                ETHBalance: account.balance
+            });
+    }
+
+    // returns share price information.
+    function getSharePrice()
+        internal
+        view
+        returns (
+            uint256 sharePrice,
+            uint256 totalPooledEther,
+            uint256 totalShares
+        )
+    {
+        // The following section is left to match Renzo's implementation:
+        //*********/
+        // Get the total TVL priced in ETH from restakeManager
+        (, , uint256 totalTVL) = RESTAKE_MANAGER.calculateTVLs();
+
+        // Get the total supply of the ezETH token
+        uint256 totalSupply = EZETH.totalSupply();
+
+        // Return the rate
+        uint256 rate = (10 ** 18 * totalTVL) / totalSupply;
+        //*********/
+
+        // My implementation
+        // (, , totalPooledEther) = RESTAKE_MANAGER.calculateTVLs();
+        // totalShares = EZETH.totalSupply();
+        // sharePrice = totalPooledEther.divDown(totalShares);
+
+        return (rate, totalTVL, totalSupply);
```

Comment:
We should just call `renzoOracle.calculateMintAmount` with `ONE` provided, since that's what Hyperdrive does. I think the comments make sense as a way of getting to the answer, but I don't think we need them when we merge.

Reply:
Roger, calling the renzo oracle's calculateRedeemAmount here, which is what _pricePerVaultShare calls under the hood now.